instrumentation/spring/spring-batch-3.0/javaagent/src/test/groovy/ItemLevelSpanTest.groovy

```diff
@@ -255,45 +261,41 @@ abstract class ItemLevelSpanTest extends AgentInstrumentationSpecification {
           kind INTERNAL
           childOf span(14)
         }
-
-        // empty chunk on thread 2, end processing
-        span(19) {
-          name "BatchJob parallelItemsJob.parallelItemsStep.Chunk"
-          kind INTERNAL
-          childOf span(1)
-        }
-        // end of stream marker
-        span(20) {
-          name "BatchJob parallelItemsJob.parallelItemsStep.ItemRead"
-          kind INTERNAL
-          childOf span(19)
-        }
-
-        // empty chunk on thread 1, end processing
-        span(21) {
-          name "BatchJob parallelItemsJob.parallelItemsStep.Chunk"
-          kind INTERNAL
-          childOf span(1)
-        }
-        // end of stream marker
-        span(22) {
-          name "BatchJob parallelItemsJob.parallelItemsStep.ItemRead"
-          kind INTERNAL
-          childOf span(21)
-        }
       }
     }
   }
+
+  def postProcessParallelItemsJob(String jobName, Job job) {
+    if ("parallelItemsJob" == jobName) {
+      Step step = ((AbstractJob) job).getStep("parallelItemsStep")
+      TaskletStep taskletStep = (TaskletStep) step
+      RepeatTemplate stepOperations = new RepeatTemplate()
+      // explicitly set the number of chunks we expect from this test to ensure we always get
+      // the same number of spans
+      stepOperations.setCompletionPolicy(new SimpleCompletionPolicy(3))
```

Comment:
That'll override the default `TaskExecutorRepeatTemplate` that's responsible for running things in parallel; can we set the completion policy without overwriting the thread pool? E.g.
```groovy

  @Bean
  Step parallelItemsStep() {
    def repeatTemplate = new TaskExecutorRepeatTemplate()
    repeatTemplate.taskExecutor = asyncTaskExecutor()
    repeatTemplate.throttleLimit = 2
    repeatTemplate.completionPolicy = new SimpleCompletionPolicy(3)

    steps.get("parallelItemsStep")
      .chunk(2)
      .reader(new TestSyncItemReader(5))
      .processor(itemProcessor())
      .writer(itemWriter())
      .stepOperations(repeatTemplate)
      .build()
  }
```